cmake/TestSetup.cmake

```diff
@@ -11,6 +11,14 @@
 # the public, perform publicly and display publicly, and to permit others to do so.
 #=========================================================================================
 
+# Seaches for the python interpreter
+message(STATUS "Searching for python 3.6, required to run regression tests.")
+include(FindPython)
```

Comment:
I don't think this is necessary. I'm actually not even sure what this would do. `find_package(Foo)` looks for a module called `FindFoo` and invokes that, so the `find_package(Python3)` on the next line is sufficient.

Reply:
They don't do the same thing. include(FindPython) exposes several variables that are otherwise hidden. It is not an old feature and is currently why the ci is failing because the module is only available after cmake version 12 something.

Reply:
@JoshuaSBrown Is this ready to merge into trunk?

Reply:
No, I'm working on it right now, your recent merge, does not appear to correctly use the PythonModuleCheck.cmake, because you removed the inlcude(FindPython), I'm adding a conditional statement to deal with this if cmake is less than 3.12.

Reply:
Seeing your other PR, I think it's because `find_package(Python3 ...)` doesn't set the the `Python_` variables, only the `Python3_` variables, but I think that all the result variables you need are there: https://cmake.org/cmake/help/latest/module/FindPython3.html?highlight=findpython3#result-variables

Reply:
Oh, is that what's going on, @AndrewGaspar. Ok, then some of the variables in the python module in the /cmake/ folder need to be updated.

Reply:
This variable does not appear to be defined using just find_package(Python3 ...) blah blah blah. At least not for cmake 3.11. The problem appears to be the cmake versions. find_package(Python3 ... ) only defines the needed Python3 variables after cmake 3.13, whereas for the Python variables they are defined for cmake 3.12. We either need to update the cmake version or specifically define these variables, or a workaround.

Reply:
Have we considered simply throwing ourselves into the sea?